### PR TITLE
bugfix + tests

### PR DESCRIPTION
**Pull request by mersinvald**

### CODE REVIEW

File: src/backend/tests/path_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
+package tests
+
+import (
+	"testing"
+	"../calc"
+	"math/big"
+	"strings"
+)
+
+func toBigingSlice(strSlice []string) ([]*big.Int, bool) {
```
Comment (ghost) [inline src/backend/tests/path_test.go]: ‘Biging’? Probably typo — ‘bigint’?

**Comment by mersinvald** [comment]
@Aljutor reference the code when you'are commenting, I can't understand you

File: src/backend/tests/path_test.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+package tests
+
+import (
+	"testing"
+	"../calc"
+	"math/big"
+	"strings"
+)
+
+func toBigintSlice(strSlice []string) ([]*big.Int, bool) {
```
Comment (Aljutor) [inline src/backend/tests/path_test.go]: mb use camelCase ? like toBigIntSlice

**Comment by mersinvald** [comment]
@Aljutor more comments or it is OK to merge now?

**Comment by Aljutor** [comment]
@mersinvald As initial ok, but better to fully cover specs by @NIMFETRISA

**Review by Aljutor — APPROVED** [review]

File: src/backend/tests/path_test.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		strs = append(strs, a[i].String())
+	}
+	strs = append(strs, "]")
+
+	return strings.Join(strs, "")
+}
+
+func printExpectedGotBigintSlices(t *testing.T, a, b []*big.Int) {
+	t.Error("assertion a == b failed")
+	t.Error("expected length ", len(a), " got lenght ", len(b))
```
Comment (ghost) [inline src/backend/tests/path_test.go:38]: 'got leng**th**'

**Review by ghost — COMMENTED** [review]
A few more comments.

1) What if input contains letters or other symbols? I saw that the input box in the front-end is bound to numbers, but it's still possible to send a POST-request with rubbish there. Leaves space for potential security breach.

2) What if a floating point number is given as input?

3) It might be valuable to benchmark a sequence starting with a power of 2 (say, 2^16) against a few other numbers, that have the same sequence length, but include '*3+1' steps as well.

**Comment by mersinvald** [comment]
@gingerDevilish 
> 1. What if input contains letters or other symbols?
> 2. What if a floating point number is given as input?

https://github.com/DevHeap/collatz_conjecture/blob/7dc28983a8544620f33c4980b0b056f8b1f81a1e/src/backend/server/websocket.go#L81-L85

> 3. It might be valuable to benchmark a sequence starting with a power of 2 

It is just one particular case, we do not distinguish power of 2 numbers in Calc from the other ones. I don't see a point in this kind of benchmark, it will show nothing considering the app logic, where ^2 numbers are not a big part of whole calculation sequences.

**Pull request merged**